test/expected/partitionwise-13.out

```diff
@@ -0,0 +1,884 @@
+                     ->  Seq Scan on public.pg2dim_h2_t2 pg2_6
+                           Output: pg2_6."time", pg2_6.device, pg2_6.temp
+(33 rows)
+
+-- Test hypertable with time partitioning function
+CREATE OR REPLACE FUNCTION time_func(unixtime float8)
+    RETURNS TIMESTAMPTZ LANGUAGE PLPGSQL IMMUTABLE AS
+$BODY$
+DECLARE
+    retval TIMESTAMPTZ;
+BEGIN
+    retval := to_timestamp(unixtime);
+    RETURN retval;
+END
+$BODY$;
+CREATE TABLE hyper_timepart (time float8, device int, temp float);
+SELECT * FROM create_hypertable('hyper_timepart', 'time', 'device', 2, time_partitioning_func => 'time_func');
+NOTICE:  adding not-null constraint to column "time"
+ hypertable_id | schema_name |   table_name   | created 
+---------------+-------------+----------------+---------
+             3 | public      | hyper_timepart | t
+(1 row)
+
+-- Planner won't pick push-down aggs on table with time function
+-- unless a certain amount of data
+SELECT setseed(1);
+ setseed 
+---------
+ 
+(1 row)
+
+INSERT INTO hyper_timepart
+SELECT x, ceil(random() * 8), random() * 20
+FROM generate_series(0,5000-1) AS x;
+-- All partition keys covered (full partitionwise)
+SET enable_partitionwise_aggregate = 'off';
+:PREFIX
+SELECT time, device, avg(temp)
+FROM hyper_timepart
+GROUP BY 1, 2
+ORDER BY 1, 2
+LIMIT 10;
+                                                QUERY PLAN                                                 
+-----------------------------------------------------------------------------------------------------------
+ Limit
+   Output: _hyper_3_7_chunk."time", _hyper_3_7_chunk.device, (avg(_hyper_3_7_chunk.temp))
+   ->  Sort
+         Output: _hyper_3_7_chunk."time", _hyper_3_7_chunk.device, (avg(_hyper_3_7_chunk.temp))
+         Sort Key: _hyper_3_7_chunk."time", _hyper_3_7_chunk.device
+         ->  HashAggregate
+               Output: _hyper_3_7_chunk."time", _hyper_3_7_chunk.device, avg(_hyper_3_7_chunk.temp)
+               Group Key: _hyper_3_7_chunk."time", _hyper_3_7_chunk.device
+               ->  Append
+                     ->  Seq Scan on _timescaledb_internal._hyper_3_7_chunk
+                           Output: _hyper_3_7_chunk."time", _hyper_3_7_chunk.device, _hyper_3_7_chunk.temp
+                     ->  Seq Scan on _timescaledb_internal._hyper_3_8_chunk
+                           Output: _hyper_3_8_chunk."time", _hyper_3_8_chunk.device, _hyper_3_8_chunk.temp
+(13 rows)
+
+:PREFIX
+SELECT time_func(time), device, avg(temp)
+FROM hyper_timepart
+GROUP BY 1, 2
+ORDER BY 1, 2
+LIMIT 10;
+                                                                     QUERY PLAN                                                                      
+-----------------------------------------------------------------------------------------------------------------------------------------------------
+ Limit
+   Output: (time_func(_hyper_3_7_chunk."time")), _hyper_3_7_chunk.device, (avg(_hyper_3_7_chunk.temp))
+   ->  GroupAggregate
```

Comment:
I see some of the HashAggregate became GroupAggregates (for thsi file) in PG13. Do we need to set enable_hashagg = false here to ensure that plans will not change?

Reply:
i think the groupagg plans are preferable and i think we dont care what type of aggregation postgres uses here and the test is already pg-version specific so it should be stable